static/js/composebox_typeahead.js

```diff
@@ -129,18 +129,21 @@ function handle_keydown(e) {
                 nextFocus = false;
             }
 
-            // Send the message on Ctrl/Cmd-Enter or if the user has configured enter to
+            // Send the message on Ctrl/Cmd-Enter, Shift/Cmd-Enter
+            // Send message if the user has configured enter to
             // send and the Shift/Ctrl/Cmd/Alt keys are not pressed.
-            // Otherwise, make sure to insert a newline instead
+            // Otherwise, add a newline for Shift/Ctrl/Cmd
             if (e.target.id === "new_message_content" && code === 13) {
-                if ((!page_params.enter_sends && (e.metaKey || e.ctrlKey)) ||
+                if ((!page_params.enter_sends && (e.metaKey || e.ctrlKey || e.shiftKey)) ||
                     (page_params.enter_sends && !(e.shiftKey || e.ctrlKey || e.metaKey || e.altKey))
                 ) {
                     e.preventDefault();
                     if ($("#compose-send-button").attr('disabled') !== "disabled") {
                         $("#compose-send-button").attr('disabled', 'disabled');
                         compose.finish();
                     }
+                } else if (page_params.enter_sends && (e.ctrlKey || e.metaKey)) {
+                    $("#new_message_content").val( $("#new_message_content").val() + "\n");
```

Comment:
`$("#new_message_content").val($("#new_message_content").val() + "\n");` (remove the space)

Not sure if we wanted Meta key to trigger a new line though, so you might want to remove the metaKey from the `else if` conditional; can you confirm @rishig?

Reply:
interesting. I'm not sure; a good question for #design.

Reply:
@synicalsyntax @rishig so what should be the final call? Should I remove the Meta key trigger?